compat/pipe.h

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#ifndef SQUID_COMPAT_PIPE_H
+#define SQUID_COMPAT_PIPE_H
+
+#if HAVE_UNISTD_H
+#include <unistd.h>
```

Comment:
What is this header included for?
 * MSVC situation: It does not exist.
 * MinGW situation: **if** it is somehow detected, then `windows.h` include will break compile.

Reply:
The idea is to have units requiring pipe(2) include compat/pipe.h , regardless of the OS